docs/DeveloperGuide.md

```diff
@@ -1,29 +1,265 @@
+application when a user inputs a command `work`.
+
+For `CommandFactory` and `Minigame`,
+* Each defines its API for creating commands and mini-games respectively, where
+  `Command` is the API for `CommandFactory` and `MiniGame` is the API for `MiniGame`.
+* Implements its functionality with concrete classes such as `WorkCommand` and `TypingGame`.
+
+The sections below would give more details of each component.
+
+## UI components
+
+UI consists of the following components:
+* `Parser`
+* `ResponseManager`
+* `CommandType`
+
+![UI.png](UML%20diagram%2FUI.png)
+
+The **UI components**,
+* `Parser` parses the user input for `CommandFactory` to produce `Command`
+* `ResponseManager` generate the response to the user according to the *command execution* and *game logic*.
+
+## EconoCraftLogic components
+
+Here is the partial class diagram of the `EconoCraftLogic` component:
+
+![Logic.png](UML%20diagram%2FLogic.png)
+
+> [!NOTE]
+> - The `XYZ` in `XYZCommand` represents the exact command e.g., `WorkCommand`, `RestCommand`, `ExerciseCommand`.
+> - The `XYZ` in `XYZMiniGame` represents the exact mini-game e.g., `TypingGame`, `TicTacToe`, etc.
+> - The `XYZ` in `StockXYZ` represents the stock label e.g., `StockOne`, `StockTwo`, etc.
+
+Here is the sequence diagram of the `EconoCraftLogic` executing the command:
+
+![Work.png](UML%20diagram%2FWork.png)
+
+The `EconoCraftLogic` mechanism:
+
+1. `EconoCraftLogic` receives the user input string and pass it into `CommandFactory`.
+2. `CommandFactory` would use `Parser` to parse the user input string and produce a `Command`.
+3. `EconoCraftLogic` would execute the `Command` which will:
+   * update the player profile accordingly.
+   * use `ResponseManager` to generate the response to the user.
+
+Here is the sequence diagram of `stock` command:
 
-{Describe the design and implementation of the product. Use UML diagrams and short code snippets where applicable.}
+![Stock.jpeg](UML%20diagram%2FStock.jpeg)
 
+The mechanism:
+1. `stock` command invoke the `StockCommand` class.
+2. `StockCommand` invokes the `start` method for `StockActivate` class.
+3. `StockActivate` class calls a new `StockStorage` class.
+4. `StockStorage` class will new execute its play function and enable user to buy stocks.
+
+Here is the sequence diagram of `sellstock` command:
+
+![SellStock.jpeg](UML%20diagram%2FSellStock.jpeg)
+
+The mechanism:
+1. `sellstock` command invoke the `SellStockCommand` class.
+2. `SellStockCommand` invokes the `getAsset` method for `PlayerProfile` and fetch the `Asset` class.
+3. `Asset` class executes the `sellStock()` function and sell all of the stocks the player currently possess.
+4. Information related to stocks will be updated inside `Asset` and `PlayerProfile` class.
+
+![Save.png](UML%20diagram%2FSave.png)
+
+The mechanism:
+1. `Saver` will be called automatically after one command is executed.
```

Comment:
Perhaps consider adding a screenshot of an example saved file, so the readers can visualise it better.